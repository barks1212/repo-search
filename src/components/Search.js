import React from 'react';

class Search extends React.Component {
  state = {
    input: '',
    repos: null,
    repoCount: null
  };

  render() {
    const { repos, repoCount } = this.state;
    return (
      <section className="search">
        <section className="field">
          <p className="control has-icons-left">
            <input className="input is-medium is-rounded is-info" onChange={this.inputHandler}
              type="text" value={this.state.input} placeholder="eg: Hangman" />
            <span className="icon is-small is-left">
              <i className="fas fa-search"></i>
            </span>
          </p>
          <button className="button is-medium is-info is-rounded" onClick={() => this.repoSearch(this.state.input)}>Search</button>
          {repoCount === 0 ?
            <NoReposFound />
            :
            <Repos />
          }
        </section>
      </section>
    );
  }

  inputHandler = (event) => {
    this.setState({ input: event.target.value });
  };

  repoSearch = (searchVal) => {
    if (searchVal === '') return;
    fetch(`https://api.github.com/search/repositories?q=${searchVal}`)
      .then(res => {
        return res.json();
      }).then(repos => {
        let top20 = repos.items.slice(0, 20).sort((a, b) => {
          return b.forks_count - a.forks_count;
        });
        this.setState({
          repos: top20,
          input: '',
          repoCount: repos.total_count
        });
      });
  };
}

export default Search;