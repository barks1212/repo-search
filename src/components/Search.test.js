import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import React from 'react';
import renderer from 'react-test-renderer';
import Search from './Search';

describe('<Search>', () => {
  it('renders as expected without crashing and matches snapshot', () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});