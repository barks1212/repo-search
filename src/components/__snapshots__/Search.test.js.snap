// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> renders as expected without crashing and matches snapshot 1`] = `
<section
  className="search has-text-centered-mobile"
>
  <section
    className="field"
    id="searchControls"
  >
    <p
      className="control has-icons-left"
      id="searchBar"
    >
      <input
        className="input is-rounded is-info"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="eg: Hangman"
        type="text"
        value=""
      />
      <span
        className="icon is-small is-left"
      >
        <i
          className="fas fa-search"
        />
      </span>
    </p>
    <button
      className="button is-info is-rounded"
      id="searchButton"
      onClick={[Function]}
    >
      Search
    </button>
  </section>
</section>
`;
